protocol/switch.go

```diff
@@ -0,0 +1,27 @@
+package protocol
+
+import (
+	"io"
+)
+
+// HandlerFunc is a user-provided function used by the Router to
+// handle a protocol/stream.
+type HandlerFunc = func(protocol string, rwc io.ReadWriteCloser) error
+
+type Router interface {
```

Comment:
Please add a comment to interfaces in this file.

Reply:
Iteration 2 cc @yusefnapora.